app/assets/stylesheets/components/syntax.scss

```diff
@@ -34,6 +34,94 @@ div.highlight {
   }
 }
 
+div.highlight {
+  position: relative;
+
+  // this a way to separate scroll of long code & icons in panel
+  .highlight {
+    overflow: auto;
+  }
```

Comment:
There's one last thing we could try to fix... So this style changes a little bit the behavior of how overflow looks like.. On prod, when you have long lines in `highlight` the code is actually from edge to edge when scrolling. 

Because of that additional styling here ^ there's an offset on the left and right side, defined by the padding, so the code is no longer from edge to edge.

I do understand why you have this styling here (so the buttons are sticked to the corner of the code block), although maybe we could try to bring the previous styling back (so code will be edge to edge when scrolling). I think the trick here is to move the padding from the outer `.highlight` to the inner one. But this may need a bit more testing though.

If you need help with it, I'm happy to jump in.

Reply:
@ludwiczakpawel,  sure, I will check it!

Reply:
I move padding to inner element -  .highlight, where overflow is and it worked. Now we also have proper calculation of paddings (screenshot will be attached) for wide scrollable area (before text had no right padding at the end of the line).